src/lib/generics-events-subscribers/erc20-event-subscriber.ts

```diff
@@ -0,0 +1,202 @@
+import _ from 'lodash';
+import { LogDescription } from '@ethersproject/abi';
+import { DeepReadonly } from 'ts-essentials';
+import { IDexHelper } from '../../dex-helper';
+import { StatefulEventSubscriber } from '../../stateful-event-subscriber';
+import { Log, BlockHeader, Token, Address } from '../../types';
+import { erc20Iface } from '../utils-interfaces';
+import {
+  decodeERC20Transfer,
+  decodeWrappedDeposit,
+  decodeWrappedWithdrawal,
+} from './utils';
+import { BalanceRequest, getBalances } from '../tokens/balancer-fetcher';
+import {
+  AssetType,
+  DEFAULT_ID_ERC20,
+  DEFAULT_ID_ERC20_AS_STRING,
+} from '../tokens/types';
+import { ERC20Event, ERC20StateMap, WrappedEvent } from './types';
+
+export const handleTransferEvent = (
+  event: LogDescription,
+  state: ERC20StateMap,
+) => {
+  const erc20Transfer = decodeERC20Transfer(event);
+
+  if (erc20Transfer.from in state) {
+    state = _.cloneDeep(state);
```

Comment:
It is not allowed to directly modify state without making cloneDeep?
I remember we were doing that, but maybe it is changed since last time I saw it

Reply:
No it's not it will corrupt in case of reorg